Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs

```diff
@@ -301,9 +301,16 @@ private void OnEnd(object exception, object thisObj, object returnValue)
                 {
                     this.TelemetryTable.Remove(thisObj);
                     DependencyTelemetry telemetry = telemetryTuple.Item1;
-                    
+
                     var responseObj = returnValue as HttpWebResponse;
 
+                    var webException = exception as WebException;
+
+                    if (webException != null)
+                    {
+                        responseObj = webException.Response as HttpWebResponse;
+                    }
```

Comment:
We can also get Http status code from HttpException ([more info](https://msdn.microsoft.com/en-us/library/system.web.httpexception.gethttpcode(v=vs.110).aspx)). Should we attempt to do it in case we are dealing with HttpException and don't have response?